src/Online_Bookstore/Dashboard/BookDetail.js

```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+import useFetch from "../CustomHooks/useFetch";
+import BookDetailView from "./BookDetailView";
+import {handleSubmit} from './SetReviewInLocalStorage'
+const BookDetail = () => {
+  const [comment, setComment] = useState("");
+  const [rating, setRating] = useState(0);
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const{booksData}=useFetch(`https://api.itbook.store/1.0/books/${id}`)
+  handleSubmit({id,rating,booksData,comment})
+  useEffect(() => {
+    if (!window.localStorage.getItem("user_login")) {
+      navigate("/");
+    }
+  }, []);
+  return (
+    <>
```

Comment:
Avoid adding extra nodes to the DOM by using React.Fragment (or, <></>), when you have already wrapped the **BooksDetailView** component in a **div** and **BooksDetailView** component does not have any other sibling component rendered together.

@Poorvi-3pillar 

```
return (
  <div>
    <BooksDetailView /> 
  </div>
)
```